python_modules/dagster/dagster/_check/builder.py

```diff
@@ -313,6 +315,12 @@ def build_check_call_str(
             return f'check.mapping_param({name}, "{name}", {_name(pair_left)}, {_name(pair_right)})'
         elif origin is collections.abc.Set:
             return f'check.set_param({name}, "{name}", {_name(single)})'
+        # If the type is a Generic record, we just use an inst check
+        elif hasattr(origin, _RECORD_MARKER_FIELD):
+            it = _name(ttype)
+            return (
+                f'{name} if isinstance({name}, {it}) else check.inst_param({name}, "{name}", {it})'
+            )
```

Comment:
@OwenKephart in this PR has a fix for `Generic` in general which may be preferable 
https://github.com/dagster-io/dagster/pull/24629/files#r1771573736

Reply:
Owen's case handles the nested generics well, but doesn't handle optional generics - so updated this PR to only handle that case https://github.com/dagster-io/dagster/pull/24629/files#diff-7c1c0348fb90d94443f1ed84746f7756b4dd57b669b817328e7ca2eff1faf6c8R369